compiler/src/main/kotlin/software/amazon/lastmile/kotlin/inject/anvil/processor/ContributesBindingProcessor.kt

```diff
@@ -118,17 +118,24 @@ internal class ContributesBindingProcessor(
                                     }
                                 }
                                 .apply {
-                                    val parameterName = clazz.innerClassNames().decapitalize()
-                                    addParameter(
-                                        ParameterSpec
-                                            .builder(
-                                                name = parameterName,
-                                                type = clazz.toClassName(),
-                                            )
-                                            .build(),
+                                    val constructor = clazz.primaryConstructor!!
```

Comment:
I'm pretty new to KSP so wasn't entirely sure how to test this, but did manage to get this to compile by generating the parameters with the `@Assisted` annotation.

Reply:
Could you please add a test in the `:compiler` module? There are already many tests that should guide you.